test/test_utils/eks.py

```diff
@@ -215,10 +228,29 @@ def delete_cfn_stack_and_wait(stack_name):
         LOGGER.error(f"Error: Cannot delete stack: {stack_name}. Full Exception:\n{e}")
         describe_cfn_stack_events(stack_name)
 
+def delete_oidc_provider(eks_cluster_name):
+    """Function to delete the oidc provider created by kubeflow
+    Args:
+        eks_cluster_name: str
+    """
+    eks_client = init_eks_client()
+    iam_client = init_iam_client()
+    account_id = os.getenv("ACCOUNT_ID")    
+
+    try:
+        response = eks_client.describe_cluster(name=eks_cluster_name)
+        oidc_issuer = response['cluster']['identity']['oidc']['issuer']
+        oidc_url = oidc_issuer.rsplit('//', 1)[-1]
+        oidc_provider_arn = f"arn:aws:iam::{account_id}:oidc-provider/{oidc_url}"
+
+        LOGGER.info(f"Deleting oidc provider: {oidc_provider_arn}")
+        iam_client.delete_open_id_connect_provider(OpenIDConnectProviderArn=oidc_provider_arn)
+    except ClientError as e:
+        LOGGER.error(f"Error: Cannot delete the EKS cluster: {eks_cluster_name}. Full Exception:\n{e}")
```

Comment:
Could you make this LOGGER.error message more specific, and say that the OIDC provider could not be deleted, followed by the arn of the provider that couldn't be deleted?

Reply:
Delete oidc provider call is idempotent; it does not fail or return an error if you call the operation for a provider that does not exist.